apps/artemis-web/components/bgp-table/bgp-table.tsx

```diff
@@ -392,36 +392,77 @@ const columns = [
 const BGPTableComponent = (props) => {
   let bgp;
   const bgpData = props.data;
-  const asns = props.asns ?? [];
-  const [ASNTitle, setASNTitle] = React.useState({});
+  const context = React.useContext(TooltipContext);
+  const [tooltips, setTooltips] = useState({});
 
   if (bgpData && bgpData.length) {
     bgp = props.data.map((row, i) => {
       const origin_as = (
         <>
-          <div data-tip data-for={'origin' + i}>
+          <div
+            onMouseOver={() =>
+              fetchTooltip(row['origin_as'], context, {
+                tooltips: tooltips,
+                setTooltips: setTooltips,
+              })
+            }
+            data-tip
+            data-for={'origin' + i}
+          >
             {row['origin_as']}
           </div>
-          <ReactTooltip html={true} id={'origin' + i}>
-            {ASNTitle[row['origin_as']] ?? 'Loading...'}
+          <ReactTooltip
+            place="top"
+            effect="solid"
+            html={true}
+            id={'origin' + i}
+          >
+            {tooltips[row['origin_as']] ?? 'Loading...'}
           </ReactTooltip>
         </>
```

Comment:
I think all this code can become another TSX component that you pass only the variables that change so you can avoid rewriting this over and over again!